interface.py

```diff
@@ -0,0 +1,258 @@
+    # print(("## YOUR CHOICE: "))
+    # x = input("@$@#%^@%@##@$%#^*&^  WORK IN PROGRESS... Press ENTER to go back to your budget.")
+
+    add_transaction()
+
+def add_transaction():
+    transaction_name = input("Write name of transaction: ")
+    transaction_payee_name = input("Write name of payee/payer: ")
+    transaction_amount_inflow = float(input("Inflow amount: "))
+    transaction_amount_outflow = float(input("Outflow amount: "))
+    transaction_category_id = int(input("Write category id: "))
+
+    transaction = Transaction(name=transaction_name,
+                                  payee_name=transaction_payee_name,
+                                  amount_inflow=transaction_amount_inflow,
+                                  amount_outflow=transaction_amount_outflow,
+                                  category_id=transaction_category_id)
+    session.add(transaction)
+    session.commit()
+
+    print_budget()
+
+
+def edit_categories():
+    print("\nEDIT CATEGORIES MENU:")
+    print("   PARENT CATEGORY:")
+    print("      1. Add parent category")
+    print("      2. Remove parent category   [FUNCTION NOT AVAILABLE YET]")
+    print("      3. Rename parent category   [FUNCTION NOT AVAILABLE YET]")
+    print("   CATEGORY (subcategory of the parent category):")
+    print("      4. Add category")
+    print("      5. Remove category   [FUNCTION NOT AVAILABLE YET]")
+    print("      6. Rename category   [FUNCTION NOT AVAILABLE YET]")
+    print("")
+    print("      7. Go back to the budget")
+    choice = input("## YOUR CHOICE: ")
+
+    if choice == "1":
+        new_parent_category = input("Write name of new parent category: ")
+        insert_into_parent_category_table = ParentCategory(name=new_parent_category, budget_id=budget_to_show)
+        session.add(insert_into_parent_category_table)
+        session.commit()
+        print_budget()
+
+    elif choice == "4":
+        new_category = input("Write name of new category: ")
+        which_parent_id = int(input("To which parent category does it belong (parent_id): "))
+        new_budgeted_amount = float(input("Write budgeted amount: "))
+
+        insert_into_category_table = Category(name=new_category, budgeted_amount=new_budgeted_amount, parent_id=which_parent_id)
+        session.add(insert_into_category_table)
+        session.commit()
+
+        # Below we need to add empty record to Transactions to avoid errors during print_budget()
+        last_category_id = list(session.query(Category.id).order_by(Category.id.desc()).first())[0]
+        empty_record = Transaction(name="# empty record to initiate a new category", amount_inflow=0, amount_outflow=0, category_id=last_category_id)
+        session.add(empty_record)
+        session.commit()
+
+        print_budget()
+
+    elif choice == "7":
+        print_budget()
+    else:
+        x = input("@$@#%^@%@##@$%#^*&^  WORK IN PROGRESS... Press ENTER to go back to your budget.")
+        print_budget()
```

Comment:
interface and logic need to be decoupled

Reply:
Tak, przydałoby się to rozbić! Tak się zastanawiam nawet, czy jest nam potrzebny plik interface - menu itp. mogłoby być definiowane w main.py, a logiki związane z klasami w controllerach.

Moglibyśmy się pozbyć wtedy jednego pośrednika.

Reply:
Ja myślę, że interface to fajny plik. Tam mogą być wszystkie funkcje printujące, które nie są generowane dynamicznie. Czyli menu(), submenu_edit_categories() (która jeszcze nie istnieje), print ascii itp.

Te funkcje mało robią (tylko printują) a zajmują bardzo dużo linijek.